src/Microsoft.AspNetCore.Blazor.Server/AutoRebuild/AutoRebuildExtensions.cs

```diff
@@ -19,7 +21,37 @@ internal static class AutoRebuildExtensions
         private static string[] _includedSuffixes = new[] { ".cs", ".cshtml", "index.html" };
         private static string[] _excludedDirectories = new[] { "obj", "bin" };
 
-        public static void UseAutoRebuild(this IApplicationBuilder appBuilder, BlazorConfig config)
+        public static void UseHostedAutoRebuild(this IApplicationBuilder appBuilder, BlazorConfig config, string hostAppContentRootPath)
+        {
+            var isFirstFileWrite = true;
+            WatchFileSystem(config, () =>
+            {
+                if (isFirstFileWrite)
+                {
+                    try
+                    {
+                        var fileToTouch = FindFirstFileInDirectoryRecursive(hostAppContentRootPath, "*.cs");
+                        if (!string.IsNullOrEmpty(fileToTouch))
+                        {
+                            File.SetLastWriteTime(fileToTouch, DateTime.Now);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        // If we don't have permission to write these files, autorebuild will not be enabled
+                        var loggerFactory = (ILoggerFactory)appBuilder.ApplicationServices.GetService(typeof(ILoggerFactory));
+                        var logger = loggerFactory?.CreateLogger(typeof (AutoRebuildExtensions));
```

Comment:
Is it actually possible for DI to not be registered? I think it's baked in at the lowest possible level in hosted ASP.NET Core.

You could also shorten this to `appBuilder.ApplicationServices.GetRequiredService<ILogger<AutoRebuildExtensions>>()` if you so desire

Reply:
Done